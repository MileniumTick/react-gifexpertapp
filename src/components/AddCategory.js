import { useState } from "react";
import propTypes from "prop-types";

export const AddCategory = ({ setCategorie }) => {

    const [input, setInput] = useState('');

    const handleInput = (e) => {
        setInput(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (input.trim().length > 2) {

            setCategorie(cat => ([input, ...cat]))
            setInput('');

        }

    }


    return <>
        <form onSubmit={handleSubmit}>
            <input type="text" value={input} onChange={handleInput} />
        </form>
    </>
}

AddCategory.propTypes = {
    setCategorie: propTypes.func.isRequired
}