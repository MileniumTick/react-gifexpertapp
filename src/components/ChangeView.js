import React from 'react'


export const ChangeView = ({origen, setOrigen}) => {

    
    const handleSticker = () => {
        setOrigen(false);
    }

    const handleGif = () => {
        setOrigen(true);
        
    }


    if (origen) {
        return <button onClick={() => handleSticker()} className="btn btn-primary">Gif</button>;
      } else {
        return <button onClick={() => handleGif()} className="btn btn-primary">Stickers</button>;
      }
  
}

