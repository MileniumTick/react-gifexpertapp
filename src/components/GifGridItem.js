import React from 'react'
import PropTypes from 'prop-types'


export const GifGridItem = ({ title, url }) => {
    return (

        /*Construye un div de cartas en bootstrap*/
        <div className="card animate__animated animate__bounceIn animate__delay-0s">
            <div className="card-body">
                <img src={url} alt={title} />
                <p>{title}</p>
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

/*
    1.Enzyme 
    2.Enzyme to json
    3.Debe de mostrar el componente correctamente
        *shallow
        *wrapper
        *wrapper .toMatchSnapshot()
*/


