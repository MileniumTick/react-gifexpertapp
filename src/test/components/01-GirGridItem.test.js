import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Prueba de <GifGridItem />', () => {

    const title = 'Titulo del Gif';
    const url = 'https://media1.giphy.com/media/3o7btLqYhQnZqgXWqc/giphy.gif';

    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);

    test('Debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de terner un parrafo con el title', () => {

        const p = wrapper.find('p').text().trim();

        expect(p).toBe(title);

    })

    test('Debe de terner una imagen con el url', () => {
            
            const img = wrapper.find('img').prop('src');
            const alt = wrapper.find('img').prop('alt');
    
            expect(img).toBe(url);
            expect(alt).toBe(title);
    
        }
    )

    test('debe tener animate_fadeIn', () => { 

        const div = wrapper.find('div').first().prop('className');

        expect(div).toContain('animate__bounceIn');
     })
});