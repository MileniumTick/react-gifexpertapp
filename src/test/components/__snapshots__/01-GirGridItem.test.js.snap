// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GifGridItem /> Debe de mostrarse correctamente 1`] = `
<div
  className="card animate__animated animate__bounceIn animate__delay-0s"
>
  <div
    className="card-body"
  >
    <img
      alt="Titulo del Gif"
      src="https://media1.giphy.com/media/3o7btLqYhQnZqgXWqc/giphy.gif"
    />
    <p>
      Titulo del Gif
    </p>
  </div>
</div>
`;
